### PR TITLE
Add new endpoint for users' run_as parameter modification

### LINKED ISSUE #6651: RBAC: RUN_AS should not be assigned during the creation of a user
Currently (Wazuh v4.0.x), it is possible to specify the "run_as" attribute during the creation of a user. This is assigning a privilege to the user and the creation of a user should not include the privileges. Privileges should be assigned using a role/policy.

This can cause the following unexpected security risk:
- Admin user creates user A. This user doesn't have "run_as", but it has the privilege to create other users.
- Then, user A creates user B with the "run_as" privilege.
- User A is skipping the "run as" limitation defined by the admin using user B.

This can be solved in this way:
- The creation of a user doesn't include the run_as privilege.
- Admin user creates the User A. It has privileges to create other users but **it doesn't have privileges to attach "run_as" to users**.

### PR DESCRIPTION
**Pull request by AdriiiPRodri**
|Related issue|
|---|
|#6651|

Hi team,

This PR closes #6651. In this PR we have added a new endpoint to enable/disable the run_as parameter of the users.

### Unittests
```
adriiiprodri@wazuh pytest -xs  wazuh/tests
=========================================== test session starts ===========================================
platform linux -- Python 3.9.1, pytest-6.2.1, py-1.10.0, pluggy-0.13.1
rootdir: /home/adriiiprodri/Desktop/git/wazuh/framework
plugins: metadata-1.11.0, html-3.1.1, tavern-1.13.1
collected 707 items                                                                                       

wazuh/tests/test_active_response.py ............
wazuh/tests/test_agent.py ....................................................................................
wazuh/tests/test_cdb_list.py ...............................................
wazuh/tests/test_ciscat.py .................................
wazuh/tests/test_cluster.py .....ss
wazuh/tests/test_decoders.py ...............................
wazuh/tests/test_group.py ........
wazuh/tests/test_logtest.py ......
wazuh/tests/test_manager.py ......................................
wazuh/tests/test_mitre.py ....................................................................................................................................................................................
wazuh/tests/test_rootcheck.py .................................................
wazuh/tests/test_rule.py ....................................................
wazuh/tests/test_sca.py .......
wazuh/tests/test_security.py .........................................................................
wazuh/tests/test_stats.py ................
wazuh/tests/test_syscheck.py ........................
wazuh/tests/test_syscollector.py ............
wazuh/tests/test_task.py ............................

========================= 705 passed, 2 skipped, 17 warnings in 76.24s (0:01:16) ==========================
```
```
adriiiprodri@wazuh pytest -xs  wazuh/core/tests 
=========================================== test session starts ===========================================
platform linux -- Python 3.9.1, pytest-6.2.1, py-1.10.0, pluggy-0.13.1
rootdir: /home/adriiiprodri/Desktop/git/wazuh/framework
plugins: metadata-1.11.0, html-3.1.1, tavern-1.13.1
collected 647 items                                                                                       

wazuh/core/tests/test_active_response.py ..............
wazuh/core/tests/test_agent.py .........................................................................................................................................................
wazuh/core/tests/test_cdb_list.py .................................
wazuh/core/tests/test_common.py ......{}
wazuh/core/tests/test_configuration.py ..................................
wazuh/core/tests/test_decoder.py ................
wazuh/core/tests/test_input_validator.py ...
wazuh/core/tests/test_logtest.py ..
wazuh/core/tests/test_manager.py ...............................
wazuh/core/tests/test_ossec_queue.py ...................
wazuh/core/tests/test_pyDaemonModule.py ......
wazuh/core/tests/test_results.py ........................................
wazuh/core/tests/test_rootcheck.py ..........
wazuh/core/tests/test_rule.py ......................
wazuh/core/tests/test_stats.py ...
wazuh/core/tests/test_syscollector.py ...
wazuh/core/tests/test_utils.py ..................................................................................................................................................................................................................
wazuh/core/tests/test_wazuh_socket.py ....................
wazuh/core/tests/test_wdb.py .....................

==================================== 647 passed, 12 warnings in 2.85s =====================================
```
```
adriiiprodri@wazuh pytest -xs  wazuh/rbac/tests             
=========================================== test session starts ===========================================
platform linux -- Python 3.9.1, pytest-6.2.1, py-1.10.0, pluggy-0.13.1
rootdir: /home/adriiiprodri/Desktop/git/wazuh/framework
plugins: metadata-1.11.0, html-3.1.1, tavern-1.13.1
collected 224 items                                                                                       

wazuh/rbac/tests/test_auth_context.py ..
wazuh/rbac/tests/test_decorators.py .........................................................................................................
wazuh/rbac/tests/test_default_configuration.py ....................................................
wazuh/rbac/tests/test_orm.py ......................................................
wazuh/rbac/tests/test_preprocessor.py ...........

=============================== 224 passed, 2 warnings in 248.96s (0:04:08) ===============================
```

### Integration tests
```
adriiiprodri@wazuh python3 run_tests.py -k security -rbac both
Collected tests [6]:
test_rbac_black_security_endpoints.tavern.yaml, test_rbac_white_security_endpoints.tavern.yaml, test_security_DELETE_endpoints.tavern.yaml, test_security_GET_endpoints.tavern.yaml, test_security_POST_endpoints.tavern.yaml, test_security_PUT_endpoints.tavern.yaml


test_rbac_black_security_endpoints.tavern.yaml 
	 25 passed, 27 warnings

test_rbac_white_security_endpoints.tavern.yaml 
	 25 passed, 27 warnings

test_security_DELETE_endpoints.tavern.yaml 
	 15 passed, 17 warnings

test_security_GET_endpoints.tavern.yaml 
	 18 passed, 20 warnings

test_security_POST_endpoints.tavern.yaml 
	 7 passed, 9 warnings

test_security_PUT_endpoints.tavern.yaml 
	 9 passed, 11 warnings
```

### CODE REVIEW

File: framework/wazuh/rbac/tests/test_orm.py [hunk 1]
```diff
@@ -306,12 +306,21 @@ def test_delete_all_policies(db_setup):
         assert len_policies == len(pm.get_policies()) + 2
 
 
+def test_enable_run_as(db_setup):
+    """Check update a user's run_as flag in the database"""
```
Comment (davidjiglesias) [inline framework/wazuh/rbac/tests/test_orm.py:310]:
```suggestion
    """Check update a user's allow_run_as flag in the database"""
```

File: framework/wazuh/security.py [hunk 2]
```diff
@@ -112,8 +112,41 @@ def get_users(user_ids: list = None, offset: int = 0, limit: int = common.databa
     return result
 
 
+@expose_resources(actions=['security:run_as'], resources=['*:*:*'])
+def enable_run_as(user_id: str = None, allow_run_as: bool = False):
+    """Enable/Disable the user's run_as flag
```
Comment (davidjiglesias) [inline framework/wazuh/security.py:117]:
```suggestion
    """Enable/Disable the user's allow_run_as flag
```

File: framework/wazuh/security.py [hunk 3]
```diff
@@ -112,8 +112,41 @@ def get_users(user_ids: list = None, offset: int = 0, limit: int = common.databa
...
+
+    Parameters
+    ----------
+    user_id : str
+        User ID
+    allow_run_as : bool
+        Enable authorization context login method for the new user
```
Comment (davidjiglesias) [inline framework/wazuh/security.py:124]:
```suggestion
        Enable or disable authorization context login method for the specified user
```

File: framework/wazuh/security.py [hunk 4]
```diff
@@ -112,8 +112,41 @@ def get_users(user_ids: list = None, offset: int = 0, limit: int = common.databa
...
+
+    Returns
+    -------
+    result : AffectedItemsWazuhResult
+        Status message
+    """
+    result = AffectedItemsWazuhResult(none_msg='The parameter run_as could not be enabled for the user',
```
Comment (davidjiglesias) [inline framework/wazuh/security.py:131]: What if we are disabling, instead of enabling? Wrong messages

Comment (davidjiglesias) [inline framework/wazuh/security.py:131]:
```suggestion
    result = AffectedItemsWazuhResult(none_msg=f"The parameter allow_run_as could not be {'enabled' if allow_run_as else 'disabled'} for the user",
```

File: framework/wazuh/security.py [hunk 5]
```diff
@@ -112,8 +112,41 @@ def get_users(user_ids: list = None, offset: int = 0, limit: int = common.databa
...
+                                      all_msg='Parameter run_as has been enabled for the user')
```
Comment (davidjiglesias) [inline framework/wazuh/security.py:132]:
```suggestion
                                      all_msg=f"The parameter allow_run_as has been {'enabled' if allow_run_as else 'disabled'} for the user")
```

File: framework/wazuh/tests/data/security/rbac_catalog.yml [hunk 6]
```diff
@@ -651,6 +651,18 @@ get_rbac_actions:
           - GET /syscollector/{agent_id}/packages
           - GET /syscollector/{agent_id}/ports
           - GET /syscollector/{agent_id}/processes
+      security:run_as:
+        description: Change the value of the run_as flag for a user
```
Comment (davidjiglesias) [inline framework/wazuh/tests/data/security/rbac_catalog.yml:655]:
```suggestion
        description: Change the value of the allow_run_as flag for a user
```

File: framework/wazuh/tests/data/security/rbac_catalog.yml [hunk 7]
```diff
@@ -651,6 +651,18 @@ get_rbac_actions:
           - GET /syscollector/{agent_id}/packages
           - GET /syscollector/{agent_id}/ports
           - GET /syscollector/{agent_id}/processes
+      security:run_as:
```
Comment (davidjiglesias) [inline framework/wazuh/tests/data/security/rbac_catalog.yml:654]:
```suggestion
      security:edit_run_as:
```

File: framework/wazuh/tests/data/security/rbac_catalog.yml [hunk 8]
```diff
@@ -651,6 +651,18 @@ get_rbac_actions:
...
+        description: Change the value of the run_as flag for a user
+        resources:
+          - "*:*"
+        example:
+          actions:
+            - security:run_as
```
Comment (davidjiglesias) [inline framework/wazuh/tests/data/security/rbac_catalog.yml:660]:
```suggestion
            - security:edit_run_as
```

File: framework/wazuh/tests/data/security/users_test_cases.yml [hunk 9]
```diff
@@ -0,0 +1,40 @@
+---
+enable_run_as:
```
Comment (davidjiglesias) [inline framework/wazuh/tests/data/security/users_test_cases.yml:2]:
```suggestion
edit_run_as:
```

**Review by davidjiglesias — CHANGES_REQUESTED** [review]
What happens with newly created wazuh and wazuh-wui users? They should have allow_run_as enabled, do they?

**Review by davidjiglesias — APPROVED** [review]

**Review by davidjiglesias — APPROVED** [review]

**Comment by vicferpoy** [comment]
## Status update
Hello team. Some of the API integration tests checks failed but they pass on local machines. We should investigate this in another issue. As for now, these are the results for those tests:

```
Collected tests [4]:
test_rbac_white_sca_endpoints.tavern.yaml, test_rbac_white_syscheck_endpoints.tavern.yaml, test_rbac_white_syscollector_endpoints.tavern.yaml, test_rbac_white_task_endpoints.tavern.yaml


test_rbac_white_sca_endpoints.tavern.yaml 
	 2 passed, 4 warnings

test_rbac_white_syscheck_endpoints.tavern.yaml 
	 4 passed, 6 warnings

test_rbac_white_syscollector_endpoints.tavern.yaml 
	 9 passed, 11 warnings

test_rbac_white_task_endpoints.tavern.yaml 
	 1 passed, 3 warnings

```

Regards,
Víctor Fernández

**Pull request merged**